synchronizer/synchronizer.go

```diff
@@ -752,7 +754,8 @@ func (s *ClientSynchronizer) checkReorg(latestBlock *state.Block) (*state.Block,
 				log.Errorf("error creating db transaction to get prevoius blocks")
 				return nil, err
 			}
-			latestBlock, err = s.state.GetPreviousBlock(s.ctx, depth, dbTx)
+			lb, err := s.state.GetPreviousBlock(s.ctx, depth, dbTx)
+			reorgedBlock = *lb
```

Comment:
I think lb can be nil here if err is not nil, right? The *lb will panic if lb is nil

Reply:
great catch!